src/main/scala/main/solutions/MazeSolver/Solver.scala

```diff
@@ -0,0 +1,33 @@
+package main.solutions.MazeSolver
+
+import scala.collection.mutable.{HashSet, Queue}
+
+object Solver {
+  def hasPathBFS(maze: Array[Array[Int]], start: Point, end: Point): Boolean = {
+    hasPathBFSHelper(maze, start, end, HashSet[Point]())
+  }
+  private def hasPathBFSHelper(maze: Array[Array[Int]], start: Point, end: Point, visited: HashSet[Point]): Boolean = {
+    val queue = Queue[Point]()
+    queue.enqueue(start)
+    while (queue.nonEmpty) {
+      val point = queue.dequeue
+      visited.add(point)
+      if (point == end)
+        return true
```

Comment:
no need for "return" statement in Scala

Reply:
I tries without it, for some reason it returned the false in the end of the function